lib/dao.js

```diff
@@ -1413,11 +1413,17 @@ DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.de
 
   var Model = this;
 
-  this.remove(byIdQuery(this, id).where, options, function(err) {
+  this.remove(byIdQuery(this, id).where, options, function(err, data) {
+    var deleted = data && data.count > 0;
     if ('function' === typeof cb) {
-      cb(err);
+      if (options.reportFailureAsError && !err && !deleted) {
```

Comment:
Perhaps `options.failWhenNotFound`, `options.mustExist` or `options.strict`.

Another approach is to use a property with `true` as a default value, e.g. `options.ignoreMissing`, `options.lenient`.